internal/services/mssql/mssql_server_security_alert_policy_resource.go

```diff
@@ -91,11 +93,9 @@ func resourceMsSqlServerSecurityAlertPolicy() *pluginsdk.Resource {
 			"state": {
 				Type:     pluginsdk.TypeString,
 				Required: true,
-				ValidateFunc: validation.StringInSlice([]string{
-					string(sql.SecurityAlertPolicyStateDisabled),
-					string(sql.SecurityAlertPolicyStateEnabled),
-					string(sql.SecurityAlertPolicyStateNew),
-				}, false),
+				// NOTE: 'SecurityAlertsPolicyStateNew' no longer exists in the new API...
+				ValidateFunc: validation.StringInSlice(serversecurityalertpolicies.PossibleValuesForSecurityAlertsPolicyState(),
```

Comment:
this is a breaking change, will existing deployments be affected by its removal in the new api or only old ones?

so should we keep the validation for the old value around till 5.0 so users existing deployments are not broken?